iree/compiler/Conversion/LinalgToSPIRV/Passes.cpp

```diff
@@ -199,43 +213,60 @@ void buildSPIRVTransformPassPipeline(OpPassManager &pm,
   //     that returns the number of workgroups.
   //   - The entry point function gets an attribute `vkspv.num_workgroups_fn` to
   //     record which function in the module returns the number of workgroups.
-  pm.addPass(createDeclareNumWorkgroupsFnPass());
+  if (!clEnableLinalgOnTensorsSPIRV) {
+    pm.nest<ModuleOp>().addPass(createDeclareNumWorkgroupsFnPass());
+  }
 
   //===--------------------------------------------------------------------===//
   // Inline the impl dispatch function into the wrapper dispatch function.
   //
   // TODO(antiagainst): re-evaluate the inlining timing.
   //===--------------------------------------------------------------------===//
-  pm.addPass(createInlinerPass());
+  pm.nest<ModuleOp>().addPass(createInlinerPass());
 
-  //===--------------------------------------------------------------------===//
-  // Inject shape calculation for output buffers.
-  //
-  // Pre-conditions:
-  //   - All transformations altering the tensor-level shapes have been done.
-  //   - "Root" dynamic tensors all pass through a single shapex.tie_shape
-  //     use which associates them to their shape.
-  //   - Loose, non-associated shapex.get_ranked_shape ops can exist anywhere
-  //     and will be resolved.
-  // Post-conditions:
-  //   - All dynamic tensors bridge through a shapex.tie_shape op with the
-  //     appropriate shape.
-  //   - No shapex.get_ranked_shape ops exist.
-  //   - Shape folding and canonicalization has been done.
-  //===--------------------------------------------------------------------===//
-  pm.addNestedPass<FuncOp>(Shape::createTieDynamicShapesPass());
-  pm.addNestedPass<FuncOp>(Shape::createMaterializeShapeCalculationsPass());
-  pm.addNestedPass<FuncOp>(Shape::createHoistShapeCalculationsPass());
+  if (clEnableLinalgOnTensorsSPIRV) {
+    addLinalgBufferizePasses(
+        pm.nest<ModuleOp>(),
+        [](OpBuilder &builder, Location loc, ArrayRef<Value> dynamicSizes,
```

Comment:
Nit: What about just creating a local variable for this lambda. Directly embedding it here makes the function call looks sad.

Reply:
Done. Hopefully it looks happier now :P